ELK_Listener/jjb_metrics_snippets.txt

```diff
@@ -0,0 +1,88 @@
+# can decide to change publish whatever you would like.
+# This is currently set up to work with the ci listener
+# from this same directory.  Essentially, it shows how
+# to manually determine the variables of interest, then
+# publish them on the message bus using the Red Hat
+# CI jenkins plugin
+
+    builders:
+     - shell: |
+        #!/bin/bash
+
+        CREATE_TIME=$(date --utc +%FT%TZ)
+        .
+        .
+        <execution of job>
+        .
+        .
+        BKR_TESTS_EXEC=0
+        BKR_TESTS_FAILED=0
+        XUNIT_LINKS=''
+        # convert all recipeSets into Junit results (do not include failed installs)
+        for recipeSet in $(cat $WORKSPACE/recipeSets.txt); do
+            bkr job-results $recipeSet | xsltproc ./kernel/jenkins/helpers/recipeSet2junit.xml - > $WORKSPACE/${{recipeSet}}_xUnit.xml
+            BKR_TESTS_EXEC=$((BKR_TESTS_EXEC+$(cat $WORKSPACE/${{recipeSet}}_xUnit.xml | grep 'failures=' | cut -d "=" -f4 | cut -d '"' -f2)))
+            BKR_TESTS_FAILED=$((BKR_TESTS_FAILED+$(cat $WORKSPACE/${{recipeSet}}_xUnit.xml | grep 'failures=' | cut -d "=" -f5 | cut -d '"' -f2)))
+            XUNIT_LINKS="$XUNIT_LINKS $JOB_URL/$BUILD_NUMBER/artifact/${{recipeSet}}_xUnit.xml"
+        XUNIT_LINKS=$(echo "$XUNIT_LINKS" | tail -c +2)
+        done
+        ARCH=''
+        for myArch in $(bkr job-results $JOBID | xsltproc ./kernel/jenkins/helpers/job2metric.xsl - 2>&1 >/dev/null | cut -d ':' -f3 | sed -e 's/^[ \t]*//' | sort | uniq) ; do
+            ARCH="$ARCH $myArch"
+        done
+        ARCH=${{ARCH:1:256}}
+        TESTS=('[{{"executor": "beaker", "arch": "'"$ARCH"'", "executed": "'$BKR_TESTS_EXEC'", "failed": "'$BKR_TESTS_FAILED'"}}]')
+        CI_TIER=0
+        if [[ -z $job_group ]] ; then OWNER=$job_group ; else OWNER=$owner ; fi
+        if [[ "$scratch" = true ]] ; then BUILD=scratch ; else BUILD=official ; fi
+        BASE_DISTRO=''
+        DISTROS=$(bkr job-results $JOBID | xsltproc ./kernel/jenkins/helpers/job2metric.xsl - 2>&1 >/dev/null | cut -d ':' -f2 | sed -e 's/^[ \t]*//' | cut -d " " -f1 | sort | uniq)
+        if [[ $(echo $DISTROS | wc -l) -eq 1 ]]; then
+              BASE_DISTRO=$DISTROS
+        else
+              for myDistro in $(echo $DISTROS) ; do
+                   BASE_DISTRO="$BASE_DISTRO $myDistro"
+              done
+        fi
+        COMPLETION_TIME=$(date --utc +%FT%TZ)
+        echo "CREATE_TIME=$CREATE_TIME" >> $WORKSPACE/CI_NOTIFIER_VARS.txt
+        echo "TESTS=$TESTS" >> $WORKSPACE/CI_NOTIFIER_VARS.txt
+        echo "CI_TIER=$CI_TIER" >> $WORKSPACE/CI_NOTIFIER_VARS.txt
+        echo "OWNER=$OWNER" >> $WORKSPACE/CI_NOTIFIER_VARS.txt
+        echo "BUILD=$BUILD" >> $WORKSPACE/CI_NOTIFIER_VARS.txt
+        echo "BASE_DISTRO=$BASE_DISTRO" >> $WORKSPACE/CI_NOTIFIER_VARS.txt
+        echo "COMPLETION_TIME=$COMPLETION_TIME" >> $WORKSPACE/CI_NOTIFIER_VARS.txt
+        echo "XUNIT_LINKS=$XUNIT_LINKS" >> $WORKSPACE/CI_NOTIFIER_VARS.txt
```

Comment:
This would be generic across all jobs in the sense that you set them so that you can later inject them which allows you to use them with the ci publisher plugin.  I suppose it could be changed to a separate script, but all the environment variables would need to be available to it.  If all jobs were set with the exact same environment variables and names, then this could be uniform and would be cleaner as a default.